gnocchi/storage/__init__.py

```diff
@@ -332,40 +335,26 @@ def get_aggregated_measures(self, metrics_and_aggregations,
                     ts.truncate(aggregation.timespan)
                 results[metric][aggregation] = ts.fetch(
                     from_timestamp, to_timestamp)
-
-        return results
-
-    def get_measures(self, metric, aggregations,
-                     from_timestamp=None, to_timestamp=None,
-                     resample=None):
-        """Get aggregated measures from a metric.
-
-        Deprecated. Use `get_aggregated_measures` instead.
-
-        :param metric: The metric measured.
-        :param aggregations: The aggregations to retrieve.
-        :param from timestamp: The timestamp to get the measure from.
-        :param to timestamp: The timestamp to get the measure to.
-        :param resample: The granularity to resample to.
-        """
-        timeseries = self.get_aggregated_measures(
-            {metric: aggregations}, from_timestamp, to_timestamp)[metric]
-
-        if resample:
-            for agg, ts in six.iteritems(timeseries):
-                timeseries[agg] = ts.resample(resample)
-
-        return {
-            aggmethod: list(itertools.chain(
-                *[[(timestamp, timeseries[agg].aggregation.granularity, value)
-                   for timestamp, value
-                   in timeseries[agg].fetch(from_timestamp, to_timestamp)]
-                  for agg in sorted(aggs,
-                                    key=ATTRGETTER_GRANULARITY,
-                                    reverse=True)]))
-            for aggmethod, aggs in itertools.groupby(timeseries.keys(),
-                                                     ATTRGETTER_METHOD)
-        }
+                if resample:
+                    results[metric][aggregation] = ts.resample(resample)
+
+            for aggmethod, aggs in\
+                    itertools.groupby(aggregations, ATTRGETTER_METHOD):
+                output[metric][aggmethod] = \
+                    list(itertools.chain(
+                        *[[(timestamp,
+                            results[metric][agg].aggregation.granularity,
+                            value)
+                           for timestamp, value
+                           in results[metric][agg].fetch
+                           (from_timestamp, to_timestamp)]
+                          for agg in sorted(aggs,
+                                            key=ATTRGETTER_GRANULARITY,
+                                            reverse=True)]))
+
+        output["rawmetrics"] = {}
+        output["rawmetrics"] = results
```

Comment:
I did this to support both single and multiple metrics aggregation. For ex: It works like get_measures to give the list of measures  for a single metric and also aggregated measures for multiple metrics, so that processor.get_measures also can operate.

Reply:
Then how will I get the list of measures when rest/api/get_measures calls the get_aggregated_measures.